bin/validate-commits.js

```diff
@@ -39,20 +39,20 @@ if (argv.installGitHooks) {
   process.exit(0);
 }
 
-run().catch((error) => {
-  log(error);
-
-  process.exit(1);
-});
+run();
 
 async function run() {
   const commits = await read(getBranch());
-  const results = await commitValidator.validate(commits);
+  const { results, output, valid } = await commitValidator.validate(commits);
 
   reporter.printReport(results);
 
-  if (!results.valid && !argv.warning) {
-    throw new Error('Commit format error!');
+  if (!valid && output !== '') {
+    log(output);
+
+    if (!argv.warning) {
```

Comment:
It is a bit weird to print half the message (the commits) with the reporter, and the other half in here... 😕 

I guess if we have a reporter, it should take care of both the list of commits and the error. What are we gaining by splitting it in 2?

Reply:
How about this... We pass the results into the reporter, and format them there. As in move [these lines](https://github.com/bigcommerce/validate-commits/pull/59/files#diff-013d02081007e5ff7159e640e26ecce0d418fba101621bdecd8a63a87ddc8dcbR56-R59) inside the reporter. That will simplify this up.